### PR TITLE
ENH: lint install_requires

### LINKED ISSUE #7: lint `install_requires` in setup.py
Packages in `qiime2` org should not declare `install_requires` in setup.py now that they are hooked up to the CI/CD system and their dependencies are stored in conda recipes. Linter should disallow `install_requires`.

### PR DESCRIPTION
**Pull request by maxvonhippel**
This PR fixes #7 by linting for `install_requires` in `setup.py` (in favor of conda dependency management)

### CODE REVIEW

**Comment by maxvonhippel** [comment]
Example:

Setup.py
````PYTHON
setup(
    name="q2-emperor",
    version=versioneer.get_version(),
    cmdclass=versioneer.get_cmdclass(),
    zip_safe=False,
    install_requires='banana'
)
````

````BASH
(q2dev) ➜  q2-emperor git:(master) ✗ python ../q2lint/q2lint.py


Package dependencies should be stored in conda recipes, not in setup.py: install_requires='banana' ...
````

Setup.py
````PYTHON
setup(
    name="q2-emperor",
    version=versioneer.get_version(),
    cmdclass=versioneer.get_cmdclass(),
    packages=find_packages(),
    install_requires={'q2_emperor': ['assets/index.html']}
    author="Yoshiki Vazquez-Baeza",
    author_email="yoshiki@ucsd.edu",
    description="Display ordination plots",
    license='BSD-3-Clause',
    url="https://qiime2.org",
    entry_points={
        'qiime2.plugins':
        ['q2-emperor=q2_emperor.plugin_setup:plugin']
    },
    package_data={'q2_emperor': ['assets/index.html']},
    zip_safe=False,
)
````

````BASH
(q2dev) ➜  q2-emperor git:(master) ✗ python ../q2lint/q2lint.py


Package dependencies should be stored in conda recipes, not in setup.py: install_requires={'q2_emperor': ['assets/index.html']} ...
````

... and if there's no issue:

Setup.py
````PYTHON
setup(
    name="q2-emperor",
    version=versioneer.get_version(),
    cmdclass=versioneer.get_cmdclass(),
    packages=find_packages(),
    author="Yoshiki Vazquez-Baeza",
    author_email="yoshiki@ucsd.edu",
    description="Display ordination plots",
    license='BSD-3-Clause',
    url="https://qiime2.org",
    entry_points={
        'qiime2.plugins':
        ['q2-emperor=q2_emperor.plugin_setup:plugin']
    },
    package_data={'q2_emperor': ['assets/index.html']},
    zip_safe=False,
)
````

````BASH
(q2dev) ➜  q2-emperor git:(master) ✗ python ../q2lint/q2lint.py
(q2dev) ➜  q2-emperor git:(master)
````

**Comment by maxvonhippel** [comment]
Checking to make sure there's a license in setup (see [issue 4](https://github.com/qiime2/q2lint/issues/4)) is trivial once this PR is accepted since this PR already reads all the keys in the setup.  Therefore I'll wait until this gets merged to implement that feature.

File: q2lint.py
Comment (jairideout) [inline q2lint.py]: We were planning to allow the following "success" cases:

- `install_requires` isn't in setup.py
- `install_requires=[]` is in setup.py

Can you update your code to accept the second case as a "success"? Most of the q2 packages specify an empty `install_requires` so we might as well support that to avoid having to update those packages.

I think you can accomplish this without regexes:

```python
if 'install_requires' in text and 'install_requires=[]' not in text:
    errors.append("Package dependencies should be stored in a conda recipe instead of setup.py `install_requires`.")
```

**Review by jairideout — CHANGES_REQUESTED** [review]
Thanks @maxvonhippel! Had a comment about accepting another "success" case (empty `install_requires`).

File: q2lint.py
Comment (maxvonhippel) [reply q2lint.py]: Yep!  Will do, thanks

**Review by jairideout — APPROVED** [review]
Looks great @maxvonhippel! Tested locally :+1:

**Comment by maxvonhippel** [comment]
Awesome, thanks @jairideout

**Comment by thermokarst** [comment]
@maxvonhippel: q2studio builds are [now failing](https://travis-ci.org/qiime2/q2studio/builds/250887270?utm_source=github_status&utm_medium=notification) because of this change. @jairideout, @ebolyen, thoughts on work-arounds?

**Comment by maxvonhippel** [comment]
Huh.. that's not good.  I'll see if I can figure it out.  @thermokarst

**Comment by maxvonhippel** [comment]
Would a check for the studio be ok?  (Albeit mega hack-y).  e.g,

````PYTHON
if ("name='q2studio'" not in text and
    'install_requires' in text and
    'install_requires=[]' not in text):
      errors.append("Package dependencies should be stored in a "
                    "conda recipe instead of setup.py "
                    "`install_requires`.")
````

.... or should we follow our own advice and put dependencies in a conda recipe in `q2studio`?  (I imagine that is an exponentially bigger can of worms ... ?)

**Comment by maxvonhippel** [comment]
Let me know if you approve of the above solution, and I'll cut a PR.  Thanks.

[The patch](https://github.com/maxvonhippel/q2lint/blob/disallow-install-requires/q2lint.py#L36)

**Comment by ebolyen** [comment]
What if we had a way to disable certain lint checks? Then it's on the ~~tool~~ repo to opt-out.

**Comment by thermokarst** [comment]
@ebolyen, I like this idea!

**Pull request merged**
